arch/arm/src/Makefile

```diff
@@ -69,6 +69,19 @@ else
   NUTTX = "$(TOPDIR)$(DELIM)nuttx$(EXEEXT)"
 endif
 
+# Additional rules for system call wrapper
+
+ifeq ($(CONFIG_BUILD_FLAT),y)
+ifeq ($(CONFIG_SCHED_INSTRUMENTATION_SYSCALL),y)
```

Comment:
Why is this restricted the FLAT build.  System calls are seldom used with the FLAT build.

Reply:
This part is used to create the additional linker command option "--wrap=\<function\>" for gnu linker.
This is needed to supersede the direct kernel function call from application into newly added system call wrapper library.
(see: https://github.com/YuuichiNakamura/nuttx-task-tracer-doc/blob/master/NuttXTaskTracerInternals.md#flat-build)

Only FLAT build requires this treatment because PROTECTED/KERNEL build already have system call stub which can insert sched_note_syscall_enter/leave.